llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3778,9 +3779,90 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpGroupFMaxNonUniformAMD>(S
 template <> Value *SPIRVToLLVM::transValueWithOpcode<OpExtInst>(SPIRVValue *const spvValue) {
   SPIRVExtInst *const spvExtInst = static_cast<SPIRVExtInst *>(spvValue);
 
-  // Just ignore this set of extended instructions
-  if (m_bm->getBuiltinSet(spvExtInst->getExtSetId()) == SPIRVEIS_NonSemanticInfo)
-    return nullptr;
+  if (m_bm->getBuiltinSet(spvExtInst->getExtSetId()) == SPIRVEIS_NonSemanticInfo) {
+    switch (spvExtInst->getExtOp()) {
+    case 1: { // DebugPrintf
+      auto &spvArgs = spvExtInst->getArguments();
+      assert(spvArgs.size() >= 1 && "printf must have at least one argument");
+      auto &formatString = m_bm->get<SPIRVString>(spvArgs[0])->getStr();
+      // Ignore printfs that don't have a matching format specifiers
+      if (formatString.find("%r") == std::string::npos)
+        return nullptr;
+
+      if (formatString.find("%ra") != std::string::npos) {
```

Comment:
I think we are free to use this extended instruction to implement our vendor-specific debugging. So I prefer to add relevant codes to a sperate handler of DebugPrintf and documents its usage with descriptive comments like `--help` print.